packages/stock-location/src/loaders/connection.ts

```diff
@@ -1,9 +1,49 @@
 import {
-  ConfigurableModuleDeclaration,
+  InternalModuleDeclaration,
   LoaderOptions,
+  MODULE_RESOURCE_TYPE,
+  MODULE_SCOPE,
 } from "@medusajs/modules-sdk"
+import { DataSource, DataSourceOptions } from "typeorm"
+
+import * as StockLocationModels from "../models"
+import { MedusaError } from "medusa-core-utils"
+import { asValue } from "awilix"
 
 export default async (
-  { configModule }: LoaderOptions,
-  moduleDeclaration?: ConfigurableModuleDeclaration
-): Promise<void> => {}
+  { options, container }: LoaderOptions,
+  moduleDeclaration?: InternalModuleDeclaration
+): Promise<void> => {
+  if (
+    moduleDeclaration?.scope === MODULE_SCOPE.INTERNAL &&
+    moduleDeclaration.resources === MODULE_RESOURCE_TYPE.SHARED
+  ) {
+    return
+  }
+
+  const dbData = options?.database as Record<string, string>
+
+  if (!dbData) {
+    throw new MedusaError(
+      MedusaError.Types.INVALID_ARGUMENT,
+      `Database config is not present at module config "options.database"`
+    )
+  }
```

Comment:
suggestion(non-blocking): Centralise error handler in a util kind of method, as this will be duplicated across modules.

Reply:
This check is specific per module, not all of them have a `database` config, some have to check the redis config, etc. Unless we use some sort of basic validation for this, like we do with request bodies.
I will think about it